active-rfcs/0000-vtu-deprecations.md

```diff
@@ -0,0 +1,188 @@
+
+This PR aims to trim down the VTU API, replacing listed methods with improved docs, more examples for all the related cases or improved testing methods.
+
+## Detailed design
+
+### Methods
+
+**Wrapper.emittedByOrder -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#emittedbyorder)
+
+Rarely used, use `emitted` instead.
+
+```js
+expect(wrapper.emitted('change')[0]).toEqual(['param1', 'param2'])
+```
+
+**get -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#get) 
+
+Recently added, throws error if nothing is matched. Will be merged into `find`.
+
+```js
+expect(wrapper.find('.notExisting')).toThrow()
+
+const element = wrapper.find('.notExisting') // will throw error
+```
+
+**is -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#is)
+
+Not that much useful. Use `element.tagName` instead. 
+
+```js
+expect(wrapper.element.tagName).toEqual('div')
+```
+
+**isEmpty -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#isempty)
+
+Use custom matcher like [jest-dom](https://github.com/testing-library/jest-dom#tobeempty) on the element
+
+```js
+expect(wrapper.element).toBeEmpty()
+```
+
+**isVisible** - [Link](https://vue-test-utils.vuejs.org/api/wrapper/#isvisible)
+
+Use custom matcher like [jest-dom](https://github.com/testing-library/jest-dom#tobevisible)
+
+```js
+expect(wrapper.element).toBeVisible()
+```
+
+**isVueInstance -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#isvueinstance)
+
+Rarely used, no benefits.
+
+**props -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#props)
+
+Anti-pattern. Test what a prop does, not it's presence or value on the wrapper.
+
+**setData -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#setdata)
+
+Anti-pattern. Use `data` mounting option to set a specific state.
+
+```js
+const wrapper = mount(Component, { 
+    data () { 
+        return { 
+            field: 'overriden'  
+        }  
+    } 
+})
+``` 
+
+If you really need to update something after it mounts, just use the instance
+
+```js
+wrapper.vm.field = 'updated field'
+```
+
+**setMethods** - [Link](https://vue-test-utils.vuejs.org/api/wrapper/#setmethods)
+
+Anti-pattern. Vue does not support arbitrarily replacement of methods, nor should VTU.
+
+If you need to stub out an action, extract the hard parts away. Then you can unit test them as well.
+
+```js
+// Component.vue
+import { asyncAction } from 'actions'
+const Component = {
+    ...,
+    methods: {
+        async someAsyncMethod() {
+            this.result = await asyncAction()
+        }
+    }	
+}
+
+// spec.js
+import { asyncAction } from 'actions'
+jest.mock('actions')
+asyncAction.mockResolvedValue({ foo: 'bar' })
+
+// rest of your test
+
+```
+**setProps -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#setprops)
+
+Overriding props after the component is mounted is a hack, which introduced lots of errors, especially with watchers in VTU Beta. 
```

Comment:
Is it really unfixable? `setProps` is really useful in my experience because it allows to make sure a component adapts to changing props while it's mounted.
Having a native API that is convenient, even if it's different from regular `mount` or `shallowMount` to expose that behavior with a parent and having direct access to the tested component and not caring about the parent, would be very helpful. It would be better to be able to keep `setProps` altogether. It also faces the same problem as `setData`, we need to wait a tick

Reply:
Yep, I thought (and still think) this API isn't ideal for most use-cases, but there is some occasions where it's useful, so it will continue to be supported.

We found a clean way to implement it and it's available in the current VTU next alpha. The bugs from beta as also mostly solved. You may now do `await wrapper.setProps` to get the re-render.